sync2/device_data_ticker_test.go

```diff
@@ -31,29 +50,31 @@ func TestDeviceTickerBasic(t *testing.T) {
 		DeviceID: "b",
 	})
 	time.Sleep(duration * 2)
-	if len(payloads) != 1 {
-		t.Fatalf("expected 1 callback, got %d", len(payloads))
+	result := payloads.clone()
```

Comment:
Why do we need to clone here, sorry?

Reply:
We'd need to lock every time we access the underlying slice, which could then lead to inconsistent results (e.g len=1 when len() is called, but then increases to len=2 later on. It's easier to snapshot the slice.